src/celeritas/ext/VecgeomParams.cc

```diff
@@ -403,5 +468,94 @@ void VecgeomParams::build_metadata()
     }();
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Function for setting and/or checking size limits for CUDA stack and heap.
+ */
+bool VecgeomParams::setup_cuda_mem_limits()
```

Comment:
This also sets up the vecgeom verbosity, *and* most critically the limit setting is done because the first call to `CudaManager::Instance` has side effects (i.e. changing the cuda memory). This should be something more like "instantiate cuda volume manager".